src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1479,6 +1482,119 @@ class Simplex_tree {
+   * diagrams into the ordinary, relative, extended+ and extended- subdiagrams. 
+   * \pre This function should be called only if this->extend_filtration() has been called first!
+   * \post The coordinates of the persistence diagram points might be a little different than the
+   * original filtration values due to the internal transformation (scaling to [-2,-1]) that is 
+   * performed on these values during the computation of extended persistence.
+   * @param[in] dgm Persistence diagram obtained after calling this->extend_filtration(), 
+   * this->initialize_filtration(), and Gudhi::persistent_cohomology::Persistent_cohomology<Simplex_tree, Field_Zp >::compute_persistent_cohomology().
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the 
+   * second one is Relative, the third one is Extended+ and the fourth one is Extended-.
+   * See section 2.2 in https://link.springer.com/article/10.1007/s10208-017-9370-z for a description of these subtypes.
+   */
+  std::vector<std::vector<std::pair<int, std::pair<Filtration_value, Filtration_value>>>> compute_extended_persistence_subdiagrams(const std::vector<std::pair<int, std::pair<Filtration_value, Filtration_value>>>& dgm){
+    std::vector<std::vector<std::pair<int, std::pair<Filtration_value, Filtration_value>>>> new_dgm(4); 
+    Filtration_value x, y;
+    Filtration_value minval_ = this->minval_;
+    Filtration_value maxval_ = this->maxval_;
+    for(unsigned int i = 0; i < dgm.size(); i++){ 
+      int h = dgm[i].first; 
+      Filtration_value px = dgm[i].second.first; 
+      Filtration_value py = dgm[i].second.second;
+      if(std::isinf(py))  continue;
+      else{
+        if ((px <= -1) & (py <= -1)){
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ + (maxval_-minval_)*(py + 2); 
+          new_dgm[0].push_back(std::make_pair(h, std::make_pair(x,y))); 
+        }
+        else if ((px >=  1) & (py >=  1)){
+          x = minval_ - (maxval_-minval_)*(px - 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2); 
+          new_dgm[1].push_back(std::make_pair(h, std::make_pair(x,y))); 
+        }
+        else {
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2);
+          if (x <= y){
+            new_dgm[2].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+          else{  
+            new_dgm[3].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+        }
+      }
+    }
+    return new_dgm;
+  }
+
+  /** \brief Extend filtration for computing extended persistence. 
+   * This function only uses the filtration values at the 0-dimensional simplices, 
+   * and computes the extended persistence diagram induced by the lower-star filtration 
+   * computed with these values. 
+   * \post Note that after calling this function, the filtration 
+   * values are actually modified. The function compute_extended_persistence_subdiagrams 
```

Comment:
With the proper quotes and/or parentheses, doxygen can generate a hyperlink to the doc of that function.

Reply:
Cool, thanks!